src/ansys/fluent/core/quantity.py

```diff
@@ -717,16 +717,98 @@ def __eq__(self, other):
     def __neg__(self):
         return Quantity(-self.value, self.unit)
 
-    def __ge__(self, other):
-        if (
-            isinstance(other, Quantity)
-            and self.get_dimensions_list() == other.get_dimensions_list()
-            and self._si_unit == other._si_unit
+    def __gt__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() != other.get_dimensions_list()
         ):
+            raise ValueError("Incompatible dimensions.")
+        elif isinstance(other, Quantity):
```

Comment:
After we have done *all* the validation at the start of each method, the rest of the method can become something quite simple (we are on safe ground). So, the right-hand side can be converted to float regardless: we don't need to check what it is.

So, each method is something like:
```
validate_matching_dimensions(other)
return float(self) <op> float(other)
```

Reply:
Done. Thank you very much.